test/stores/widgets/WidgetPermissionStore-test.ts

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { mocked } from "jest-mock";
+import { MatrixClient } from "matrix-js-sdk/src/matrix";
+import { Widget, WidgetKind } from "matrix-widget-api";
+
+import { OIDCState, WidgetPermissionStore } from "../../../src/stores/widgets/WidgetPermissionStore";
+import SettingsStore from "../../../src/settings/SettingsStore";
+import { TestSdkContext } from "../../TestSdkContext";
+import { SettingLevel } from "../../../src/settings/SettingLevel";
+
+jest.mock("../../../src/settings/SettingsStore", () => ({
+    getValue: jest.fn(),
+    setValue: jest.fn(),
+    monitorSetting: jest.fn(),
+}));
+
+describe("WidgetPermissionStore", () => {
+    let widgetPermissionStore: WidgetPermissionStore;
+    let mockClient: MatrixClient;
+    const userId = "@alice:localhost";
+    const roomId = "!room:localhost";
+    const w = new Widget({
+        id: "wid",
+        creatorUserId: userId,
+        type: "m.custom",
+        url: "https://invalid.address.here",
+    });
+    let settings = {}; // key value store
+
+    beforeEach(() => {
+        settings = {}; // clear settings
+        mocked(SettingsStore.getValue).mockImplementation((setting: string) => {
+            return settings[setting];
+        });
+        mocked(SettingsStore.setValue).mockImplementation((settingName: string,
+            roomId: string | null,
+            level: SettingLevel,
+            value: any,
+        ): Promise<void> => {
+            // the store doesn't use any specific level or room ID (room IDs are packed into keys in `value`)
+            settings[settingName] = value;
+            return Promise.resolve();
+        });
+        mockClient = {
+            getUserId: () => {
+                return userId;
+            },
+        } as unknown as MatrixClient;
```

Comment:
afaik we want to avoid casts everywhere, even in tests.
You could use `stubClient()` from the test utils here.